pkg/generator/ingress_translator.go

```diff
@@ -241,6 +268,52 @@ func (translator *IngressTranslator) translateIngress(ctx context.Context, ingre
 	}, nil
 }
 
+func (translator *IngressTranslator) createUpstreamTransportSocket(activatorCA, activatorSAN string, http2 bool) (*envoy_core_v3.TransportSocket, error) {
+	caSecret, err := translator.secretGetter(system.Namespace(), activatorCA)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch activator CA secret: %w", err)
+	}
+	var alpnProtocols string
+	if http2 {
+		alpnProtocols = "h2"
+	}
+	tlsAny, err := anypb.New(createUpstreamTLSContext(caSecret.Data[caDataName], activatorSAN, alpnProtocols))
+	if err != nil {
+		return nil, err
+	}
+	return &envoy_core_v3.TransportSocket{
+		Name: wellknown.TransportSocketTls,
+		ConfigType: &envoy_core_v3.TransportSocket_TypedConfig{
+			TypedConfig: tlsAny,
+		},
+	}, nil
+}
+
+func createUpstreamTLSContext(caCertificate []byte, activatorSAN string, alpnProtocols ...string) *auth.UpstreamTlsContext {
+	return &auth.UpstreamTlsContext{
+		CommonTlsContext: &auth.CommonTlsContext{
+			AlpnProtocols: alpnProtocols,
+			TlsParams: &auth.TlsParameters{
+				TlsMinimumProtocolVersion: auth.TlsParameters_TLSv1_2,
```

Comment:
What about `TlsParameters_TLSv1_3`? Is there a reason for picking 1.2?

Reply:
TLS v1.2 is not EOL yet.

Reply:
Doe that mean that `curl --tlsv1.3 ....` will work when `auth.TlsParameters_TLSv1_2` is used?

Reply:
Cool thanks.